README.md

```diff
@@ -7,39 +7,114 @@ MalariaGEN resource centre team.
 ## Proposing changes
 
 If you want to propose changes to this repository, e.g., add or update
-a package in environment.yml, please submit a pull request.
+a package in environment.yml, please submit a pull request. The following
+commands show a typical workflow for doing this:
+
+- If you don't already have a local clone of the binder repo, create one with:
+
+```
+git clone git@github.com:malariagen/binder.git
+```
+
+- Update your local master branch:
+
+```
+cd /path/to/local/clone/of/binder
+git checkout master
+git pull
+```
+
+- Create a new issue at https://github.com/malariagen/binder/issues
+- Create a new branch for the work you want to do, naming the branch using the issue number and a descriptive slug, e.g.:
+
+```
+git checkout -b 10-add-pymysql-package
+git push -u origin 10-add-pymysql-package
+```
+
+- Make relevant changes (e.g. to environment.yml) and then add, commit and push the changes, e.g.:
+
+```
+git add environment.yml
+git commit -m 'add pymysql package #10'
+git push
+```
+
+- Come back here to https://github.com/malariagen/binder and you should now see a "Compare & pull request" button for your new branch. Click this.
+- Assuming this issue would be resolved by the pull request, include the text "resolves #10" in the PR description (replacing "10" with whatever is the issue number being resolved). This will mean that when the PR is merged, the corresponding issue will automatically get closed.
 
 ## Usage
 
 This repo is intended to be used as a git submodule within another
-repo. For example, to use this within the malariagen/vector-ops repo,
-do:
+repo. This typically involves three distinct steps:
+
+1. Add binder as a submodule to your repo. This only needs to be done once per repo. Note that someone else might already have done this for your repo of interest (the repo will include a ```binder``` sub-directory if that is the case)
+1. Run the install script for each local clone of the repo. This needs to be done for each local clone you have, for example you might need to do this separately for one clone on your local machine and a second clone on a server.
+1. Update the binder submodule. This needs to be done every time you know there have been changes made to the binder repo, and you want to move the submodule forward within your working repo.
+
+The following sub-sections give commands for each of the above.
+
+### Add binder as a submodule
+It is recommended that you first create an issue for doing this within your repo with a title such as "Add binder". Then create a new branch, add binder submodule, and create a pull request. It is also recommend that you add 'deps' to the .gitignore file for the repo, so that once the install command has ben run (see next section), the deps directory this is installed into will not come under git tracking. For example, to use this within the malariagen/vector-ops repo, assuming your new issue is number 10,
```

Comment:
```suggestion
It is recommended that you first create an issue for doing this within your repo with a title such as "Add binder". Then create a new branch, add binder submodule, and create a pull request. It is also recommend that you add 'deps' to the .gitignore file for the repo, so that once the install command has been run (see next section), the deps directory this is installed into will not come under git tracking. For example, to use this within the malariagen/vector-ops repo, assuming your new issue is number 10,
```